subsys/bluetooth/mesh/heartbeat.c

```diff
@@ -295,28 +298,26 @@ uint8_t bt_mesh_hb_sub_set(uint16_t src, uint16_t dst, uint32_t period)
 		sub.min_hops = 0U;
 		sub.max_hops = 0U;
 		sub.count = 0U;
-		sub.period = sub.period - sub_remaining();
-		if (!k_delayed_work_cancel(&sub_timer)) {
-			notify_sub_end();
-		}
+		sub.period = 0U;
```

Comment:
We'll need to follow this line up with a PTS run, I think. I couldn't find any places in the test spec where the old behavior was required, but it could be a quirk in the PTS implementation. 

Won't let this block this PR though. 
cc @michal-narajowski

Reply:
I ran HBP and HBS tests with this PR and they are passing so this looks good to me.